packages/form-core/src/choice-group/ChoiceInputMixin.js

```diff
@@ -232,6 +231,20 @@ const ChoiceInputMixinImplementation = superclass =>
       this.__isHandlingUserInput = false;
     }
 
+    /**
+     * Override this in case of extending ChoiceInputMixin and requiring
+     * to sync differently with parent form group name
+     * Right now it checks tag name match where the parent form group tagname
+     * should include the child field tagname ('checkbox' is included in 'checkbox-group')
+     */
+    _syncNameToParentFormGroup() {
+      // @ts-expect-error not all choice inputs have a name prop, because this mixin does not have a strict contract with form control mixin
+      if (this._parentFormGroup.tagName.includes(this.tagName)) {
```

Comment:
New code, new tests?

Reply:
The passing condition is tested in the choice input mixin already (e.g. if I invert this condition, some tests fail)

I will also write a test for failing the condition (means that name does not get synced)